app/controllers/candidate_interface/continuous_applications/course_choices/concerns/duplicate_course_redirect.rb

```diff
@@ -0,0 +1,24 @@
+module CandidateInterface
+  module ContinuousApplications
+    module CourseChoices
+      module Concerns
+        module DuplicateCourseRedirect
+          extend ActiveSupport::Concern
+
+          included do
+            before_action :redirect_exisiting, only: %w[new] # rubocop:disable Rails/LexicallyScopedActionFilter
+          end
+
+        private
+
+          def redirect_exisiting
+            course = Course.find(params[:course_id])
+            return unless current_application.contains_course?(course)
+
+            redirect_to candidate_interface_continuous_applications_duplicate_course_selection_path(course.provider_id, course.id)
```

Comment:
Not really related to this PR, but not sure why this path requires `provider_id`.

Reply:
I know, it was to keep the URL structure consistent across the course selection journey. If I could have it back, I would change it.

I guess I could do it in this PR. WDYT

Reply:
I think it's better to do it in a separate PR. It's easier to review.